From ee51787cd42e88550cef12fb0bf71d3ac4d329ca Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Florian=20M=C3=BCllner?= <fmuellner@gnome.org>
Date: Tue, 4 Dec 2018 15:32:35 +0100
Subject: [PATCH 29/55] prefs: Handle translations when installed system-wide

The current code assumes that the extension is installed in the user's
home, while the build system does support system-wide installation.
Account for that by checking for the presence of a locale/ subfolder
in the extension directory, and fall back to gnome-shell's own locale
directory otherwise (assuming a common prefix between extension and
gnome-shell).

https://gitlab.gnome.org/World/ShellExtensions/desktop-icons/issues/66
---
 prefs.js | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/prefs.js b/prefs.js
index 85a1128..5408728 100644
--- a/prefs.js
+++ b/prefs.js
@@ -24,6 +24,8 @@ const GioSSS = Gio.SettingsSchemaSource;
 const ExtensionUtils = imports.misc.extensionUtils;
 const Gettext = imports.gettext;
 
+const Config = imports.misc.config;
+
 Gettext.textdomain('desktop-icons');
 
 var _ = Gettext.gettext;
@@ -52,7 +54,10 @@ function initTranslations() {
     let extension = ExtensionUtils.getCurrentExtension();
 
     let localedir = extension.dir.get_child('locale');
-    Gettext.bindtextdomain('desktop-icons', localedir.get_path());
+    if (localedir.query_exists(null))
+        Gettext.bindtextdomain('desktop-icons', localedir.get_path());
+    else
+        Gettext.bindtextdomain('desktop-icons', Config.LOCALEDIR);
 }
 
 function init() {
-- 
2.20.0

